TrackletAlgorithm/MatchCalculator.h

```diff
@@ -835,68 +832,60 @@ void MatchCalculator(BXType bx,
       goodmatch_next = true;
       projseed_next  = proj_seed;
     }
-    else if (newtracklet){ // if is a new tracklet, do not make a match because it didn't pass the cuts
-      bestmatch_next = FullMatch<FMTYPE>();
-      goodmatch_next = false;
-      projseed_next  = -1;
-    }
-    else { // if current match did not pass, but it is not a new tracklet, keep the previous best match for that tracklet
-      bestmatch_next = bestmatch;
-      goodmatch_next = goodmatch;
-      projseed_next  = projseed;
-    }
 
-    if(newtracklet && goodmatch==true) { // Write out only the best match, based on the seeding 
-      switch (projseed) {
+    if(goodmatch_next&&valid_L3) { 
```

Comment:
@bryates Can you explain why `valid_L3` is used here? This block doesn't appear to be specific to L3, but I might just not understand the algorithm.

Reply:
This is layer 3 of the priority encoder, not L3 of the barrel. I suppose it's a poor variable name.

Reply:
Ah I see. That makes a lot more sense.

Yeah, maybe you can think of a better naming scheme in the future, but that's not too important right now.